### PR TITLE
feat(weapon, driving) & refactor: Added Nightvision and Heatvision, more driving styles, plate speedometer, and refactored a lot.

### PR DESCRIPTION
**Pull request by SpaghettDev**
I am still working on this. Some stuff changed in this PR:
- Additions:
  - Added nightvision and heatvision
  - Added more driving styles and changed some hard-coded values
  - Added a plate speedometer
- Deletions:
  - Removed weapons_force_crosshairs function. (self_hud already has forcing crosshair)
- Refactors:
  - Changed C-style casts to C++ casts. Ex: ```(int)eSessionType::CREW``` to ```static_cast<int>(eSessionType::CREW)```
  - Made backend looped functions more similar
  - Changed one-line if, for, while loops from having brackets to not. Ex:
```c++
if (g->vehicle.auto_drive_style == static_cast<AutoDriveStyle>(i))
{
	ImGui::SetItemDefaultFocus();
}
```
to
```c++
if (g->vehicle.auto_drive_style == static_cast<AutoDriveStyle>(i))
	ImGui::SetItemDefaultFocus();
```
  - Some other stuff i can't seem to remember

Please leave a review if something's out of place or wrong so I can fix it.

### CODE REVIEW

**Comment by Yimura** [comment]
Did you commit this in browser again?

**Comment by SpaghettDev** [comment]
> Did you commit this in browser again?

No, I never did. It's just that I have a bad habbit of changing a lot and forgetting to try and compile.

File: src/backend/looped/weapons/custom_gun_disable_control_action.cpp [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 namespace big
 {
-	static const ControllerInputs attack_controls[] =
+	static const ControllerInputs bAttackControls[] =
```
Comment (BugisoftRSG) [inline src/backend/looped/weapons/custom_gun_disable_control_action.cpp]: The b in this variable name stands for bool, doesnt apply to this

Comment (SpaghettDev) [reply src/backend/looped/weapons/custom_gun_disable_control_action.cpp]: yeah I changed it and forgot to revert

**Comment by SpaghettDev** [comment]
Build action is failing because it can't find force_crosshairs.cpp, on my machine it compiles fine...

**Comment by GeopJr** [comment]
https://github.com/YimMenu/YimMenu/pull/688/files#diff-a9d6242afd4ffb93658b6ecbbba5bddc20d7df5ff1fd9e7d6eb4de79b4317e99L73

https://github.com/YimMenu/YimMenu/pull/688/files#diff-347488dc7b657469c3d7fcb79e8105cf752229aa02404d45ce77fe9c53f7bf58L64

https://github.com/YimMenu/YimMenu/blob/bb405bb375d117c7adeaabcbac999fd6071adaba/src/backend/looped/weapons/force_crosshairs.cpp#L7

https://github.com/YimMenu/YimMenu/blob/bb405bb375d117c7adeaabcbac999fd6071adaba/src/backend/looped/weapons/force_crosshairs.cpp#L9

(that's what the CI is complaining about)

**Comment by SpaghettDev** [comment]
ah, couldn't see it for some reason.

**Comment by iwanttodo1t** [comment]
There have been many if else single lines without curly braces, and there have been some problems with the program logic, and now I feel that omitting the curly braces is easy to make the compiler compile problematic

**Pull request closed without merging**